### ISSUE TITLE
http://openmirage.org/wiki/atom.xml feed replay

### ISSUE THREAD

**Issue by dbuenzli** [mirage/mirage-www#319]
It seems that I had a http://openmirage.org/wiki/atom.xml feed replay today.

I don't know how you generate your uuids but in any case here's a relatively fullproof way of generating uuids in atom feeds. 
1. Generate a RFC 4122 V4 (random based) root uuid for your feed uuid. Keep it preciously.
2. For each entry use a RFC 4122 V5 (sha-1 name based) that uses the uuid of the feed as the name space and the original publication date (which should never change in a feed) in RFC 3339 format as the name. 

Basically for a feed with id `$FEEDID` and an entry with original publication date `$PUB_DATE` use  `uuidtrip -sha1 -ns $FEEDID -name $PUB_DATE` as the entry id. 

As long as your system doesn't change the initial publication dates of the articles and that you didn't lose your feed uuid you should then never see replays. The only constraint is that you cannot create two articles exactly at the same time (but RFC 3339 allows subsecond precision so it shouldn't really be a problem in practice).

**Comment by lnmx** [comment]
Same here; in wiki/atom.xml, the entry ids are URLs:

    <entry><id>https://openmirage.org/wiki/weekly-2015-04-09</id>...</entry>

I would guess that the `<id>`s changed from `http://` to `https://` as part of the HTTPS rollout for mirage.io?  It looks like openmirage.org is not serving HTTPS, so the links are broken, too.

**Comment by dbuenzli** [comment]
Ouch. Using URIs as feed entry IDs it the last thing you want to use.

**Comment by lnmx** [comment]
The entry ids are back to `http://`; I believe this was a deployment issue that has since been corrected.

**Comment by dbuenzli** [comment]
Could you fix this please, the feed replays are getting annoying.

**Comment by lnmx** [comment]
So, http://openmirage.org/wiki/atom.xml is now redirecting (301) to https://mirage.io/wiki/atom.xml, and the entry IDs have changed again to reflect the new host.  Hopefully the new host/redirect scheme is stable for now, and you should not see any more replays related to that.

**Comment by dbuenzli** [comment]
Well I'd still argue that this is a completely broken way of managing feed entry ids which only makes you violate [RFC 4287 section 4.2.6](https://tools.ietf.org/html/rfc4287#section-4.2.6) repeatedly (and would again e.g. if a wiki page changes its uri).
